src/client/hooks/useProfileStates.js

```diff
@@ -0,0 +1,32 @@
+import { useContext } from 'react'
+import { BoxContext } from '../wrappers/box'
+
+const defaultValues = {
+  checkingMembership: false,
+  checkedMembershipSuccess: false,
+  checkedMembershipError: false,
+  error: '',
```

Comment:
I'm a little concerned about errors throughout the application. I feel like they change types from strings to objects. Let's keep this as-is for now, and do a big sweep of cleaning up how errors are handled throughout the application. Then we can couple it with a `useErrors` hook

Reply:
There is some possibility of type changing indeed - I tried to be careful about it, but there are many places in which 'error' is used, and it makes sense to clean up in a specific ticket.